core/chains/evm/txm/storage/inmemory_store.go

```diff
@@ -197,16 +209,21 @@ func (m *InMemoryStore) MarkConfirmedAndReorgedTransactions(latestNonce uint64)
 		if tx.Nonce == nil {
 			return nil, nil, fmt.Errorf("nonce for txID: %v is empty", tx.ID)
 		}
+		existingTx, exists := m.UnconfirmedTransactions[*tx.Nonce]
+		if exists {
+			m.lggr.Errorw("Another unconfirmed transaction with the same nonce exists. Transaction will overwritten.",
+				"existingTx", existingTx, "newTx", tx)
+		}
 		if *tx.Nonce >= latestNonce {
-			tx.State = types.TxUnconfirmed
+			tx.State = txmgr.TxUnconfirmed
 			tx.LastBroadcastAt = nil // Mark reorged transaction as if it wasn't broadcasted before
 			unconfirmedTransactionIDs = append(unconfirmedTransactionIDs, tx.ID)
 			m.UnconfirmedTransactions[*tx.Nonce] = tx
 			delete(m.ConfirmedTransactions, *tx.Nonce)
 		}
 	}
 
-	if len(m.ConfirmedTransactions) >= maxQueuedTransactions {
+	if len(m.ConfirmedTransactions) > maxQueuedTransactions {
 		prunedTxIDs := m.pruneConfirmedTransactions()
```

Comment:
Why do we need to prune Confirmed Txs?
The maxQueuedTransactions setting meant to apply only to unstarted txs in txmv1.
Is this a new config for txmv2? Also, why is it set hard-coded to 250?

Reply:
We need a way to prevent the queues from growing indefinitely (all queues), otherwise they could end up using the entire memory, resulting in the node crashing. `maxQueuedTransactions` will eventually be replaced by the existing config we have `MaxQueued` once I implement the persistence storage. To figure out the optimal value memory-wise I'd have to run extensive tests, so for now, it was easier to start with a hard-coded value that we use in most of our cases just to be on the safe side.

Reply:
Ok, as long as TXM v2 is not widely use, we are good.
But I know CCIP uses their own MaxQueued values, and expect only unstarted Txs to get pruned.